ionoscloud/resource_server_cube.go

```diff
@@ -753,31 +799,37 @@ func resourceCubeServerRead(ctx context.Context, d *schema.ResourceData, meta in
 		}
 	}
 
-	if server.Properties.BootVolume != nil {
-		if server.Properties.BootVolume.Id != nil {
-			if err := d.Set("boot_volume", *server.Properties.BootVolume.Id); err != nil {
-				diags := diag.FromErr(err)
-				return diags
-			}
-		}
-		volumeObj, apiResponse, err := client.ServersApi.DatacentersServersVolumesFindById(ctx, dcId, serverId, *server.Properties.BootVolume.Id).Execute()
-		logApiRequestTime(apiResponse)
+	inlineVolumeIds := d.Get("inline_volume_ids")
 
-		if err == nil {
-
-			volumeItem := SetCubeVolumeProperties(volumeObj)
-
-			userData := d.Get("volume.0.user_data")
-			volumeItem["user_data"] = userData
+	if inlineVolumeIds != nil {
+		inlineVolumeIds := inlineVolumeIds.([]interface{})
+		var volumes []interface{}
 
-			backupUnit := d.Get("volume.0.backup_unit_id")
-			volumeItem["backup_unit_id"] = backupUnit
-
-			volumesList := []map[string]interface{}{volumeItem}
-			if err := d.Set("volume", volumesList); err != nil {
-				diags := diag.FromErr(fmt.Errorf("[DEBUG] Error saving volume to state for IonosCloud server (%s): %w", d.Id(), err))
-				return diags
+		for i, volumeId := range inlineVolumeIds {
+			volume, apiResponse, err := client.ServersApi.DatacentersServersVolumesFindById(ctx, dcId, d.Id(), volumeId.(string)).Execute()
+			logApiRequestTime(apiResponse)
+			if err != nil {
+				return diag.FromErr(fmt.Errorf("error retrieving inline volume %w", err))
 			}
+			volumePath := fmt.Sprintf("volume.%d.", i)
+			entry := SetCubeVolumeProperties(volume)
+			userData := d.Get(volumePath + "user_data")
```

Comment:
@cbeti-ionos this is very similar to what we have for the server, please check if we can create a function that can be used for both cube and enterprise server.

Reply:
there's actually a separate PR for this, which cleans up some duplicate Volume functionality and should resolve this